### PR TITLE
[Fastlane.Swift] Fix Fastlane.Swift point values default value

### LINKED ISSUE #18433: Swift APIs that expect a `Float` don't accept an `Int` literal
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Today, only `xcov` action uses `Float` type, but this issue will propagate to any action that attempts to use the `Float` type. The issue can be reproduced as follows:

```swift
// Somewhere in your Fastfile.swift...
xcov(minimumCoveragePercentage: 0) // Or any other integer literal
```

The command above crashes with exception `'minimum_coverage_percentage' value must be a Float! Found Integer instead.` as mentioned here: https://github.com/fastlane/fastlane/pull/18358

In fact that is the default value:

image

https://github.com/fastlane-community/xcov/blob/f8904c7f000b415c7800b1c68c50e8de9dab76da/lib/xcov/options.rb#L136

So in order to make `xcov` run, you must explicitly pass this value as a Float, e.g. `0.0` instead of `0`.

Relevant code:

https://github.com/fastlane/fastlane/blob/e6fc532712772f574636492bd3a9e209344854c8/fastlane_core/lib/fastlane_core/configuration/config_item.rb#L203-L205

##### Command executed

```swift
// Somewhere in your Fastfile.swift...
xcov(minimumCoveragePercentage: 0) // Or any other integer literal
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> be fastlane test --verbose
be fastlane test --verbose
[✔] 🚀 
DEBUG [2021-03-20 04:04:13.73]: Checking if there are any plugins that should be loaded...
INFO [2021-03-20 04:04:13.89]: loading manifest: /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/swift/upgrade_manifest.json
DEBUG [2021-03-20 04:04:13.96]: No missing groups found, so we don't need to worry about adding new groups
DEBUG [2021-03-20 04:04:13.97]: Fastlane.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.114/0.9.114
DEBUG [2021-03-20 04:04:13.97]: DeliverfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.61/0.9.61
DEBUG [2021-03-20 04:04:13.97]: GymfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.64/0.9.64
DEBUG [2021-03-20 04:04:13.97]: MatchfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.58/0.9.58
DEBUG [2021-03-20 04:04:13.97]: PrecheckfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.57/0.9.57
DEBUG [2021-03-20 04:04:13.97]: ScanfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.69/0.9.69
DEBUG [2021-03-20 04:04:13.97]: ScreengrabfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.59/0.9.59
DEBUG [2021-03-20 04:04:13.97]: SnapshotfileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.53/0.9.53
DEBUG [2021-03-20 04:04:13.97]: LaneFileProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: ControlCommand.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: RubyCommand.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: RubyCommandable.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: Runner.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: SocketClient.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: SocketClientDelegateProtocol.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: SocketResponse.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: main.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: ArgumentProcessor.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: RunnerArgument.swift FastlaneRunnerAPIVersion (bundled/target): 0.9.2/0.9.2
DEBUG [2021-03-20 04:04:13.97]: Checking for new user-provided tool configuration files
INFO [2021-03-20 04:04:13.98]: FastlaneSwiftRunner project is up-to-date
DEBUG [2021-03-20 04:04:13.98]: Waiting for 5 seconds for a connection from FastlaneRunner
INFO [2021-03-20 04:04:14.23]: $ ./fastlane/FastlaneRunner lane test logMode verbose swiftServerPort 2000 > /dev/null
DEBUG [2021-03-20 04:04:14.27]: Client connected
DEBUG [2021-03-20 04:04:14.27]: received command:#<Fastlane::ActionCommand:0x00007fadcec80608 @method_name="xcov", @class_name=nil, @command_id="", @args=[named argument: output_directory, value: ./xcov_report, type: , named argument: html_report, value: true, type: , named argument: markdown_report, value: false, type: , named argument: json_report, value: false, type: , named argument: minimum_coverage_percentage, value: 0, type: , named argument: skip_slack, value: false, type: , named argument: slack_username, value: xcov, type: , named argument: slack_message, value: Your *xcov* coverage report, type: , named argument: ignore_file_path, value: ./.xcovignore, type: , named argument: include_test_targets, value: false, type: , named argument: only_project_targets, value: false, type: , named argument: disable_coveralls, value: false, type: , named argument: ideFoundationPath, value: /Applications/Xcode-12.2.app/Contents/Developer/../Frameworks/IDEFoundation.framework/Versions/A/IDEFoundation, type: , named argument: legacy_support, value: false, type: ]>
ERROR [2021-03-20 04:04:14.31]: You passed invalid parameters to 'xcov'.
ERROR [2021-03-20 04:04:14.31]: Check out the error below and available options by running `fastlane action xcov`
DEBUG [2021-03-20 04:04:14.31]: sending {"payload":{"status":"failure","failure_information":["FastlaneCore::Interface::FastlaneError:","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/config_item.rb:204:in `ensure_generic_type_passes_validation'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/config_item.rb:229:in `valid?'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/config_item.rb:195:in `verify!'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:87:in `block in verify_value_exists'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:82:in `each'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:82:in `verify_value_exists'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:66:in `initialize'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:36:in `new'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane_core/lib/fastlane_core/configuration/configuration.rb:36:in `create'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/configuration_helper.rb:8:in `parse'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/runner.rb:239:in `block in execute_action'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server_action_command_executor.rb:73:in `run'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server_action_command_executor.rb:42:in `execute'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:181:in `execute_action_command'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:176:in `process_action_command'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:123:in `handle_action_command'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:86:in `parse_and_execute_command'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:62:in `block in receive_and_process_commands'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `loop'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `receive_and_process_commands'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:158:in `listen'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:38:in `start'","/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/swift_lane_manager.rb:223:in `block in start_socket_thread'"],"failure_class":"FastlaneCore::Interface::FastlaneError","failure_message":"'minimum_coverage_percentage' value must be a Float! Found Integer instead."}}
#<Thread:0x00007fadcadf1978@/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/swift_lane_manager.rb:220 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
	9: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/swift_lane_manager.rb:223:in `block in start_socket_thread'
	8: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:38:in `start'
	7: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:158:in `listen'
	6: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `receive_and_process_commands'
	5: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `loop'
	4: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:62:in `block in receive_and_process_commands'
	3: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:81:in `parse_and_execute_command'
	2: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/command_parser.rb:12:in `parse'
	1: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse'
/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse': 809: unexpected token at '{ "commandType": "control", "command": {"command":"done"} }' (JSON::ParserError)
WARN [2021-03-20 04:04:14.31]: Lane Context:
INFO [2021-03-20 04:04:14.31]: {:LANE_NAME=>"test"}
ERROR [2021-03-20 04:04:14.31]: 809: unexpected token at '{ "commandType": "control", "command": {"command":"done"} }'
ERROR [2021-03-20 04:04:14.31]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: 809: unexpected token at '{ "commandType": "control", "command": {"command":"done"} }'

URL: https://api.github.com/search/issues?q=809%3A%20unexpected%20token%20at%20%27%7B%20%22commandType%22%3A%20%22control%22%2C%20%22command%22%3A%20%7B%22command%22%3A%22done%22%7D%20%7D%27+repo:fastlane/fastlane
Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
bundler: failed to load command: fastlane (/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/bin/fastlane)
Traceback (most recent call last):
	9: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/swift_lane_manager.rb:223:in `block in start_socket_thread'
	8: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:38:in `start'
	7: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:158:in `listen'
	6: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `receive_and_process_commands'
	5: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `loop'
	4: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:62:in `block in receive_and_process_commands'
	3: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:81:in `parse_and_execute_command'
	2: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/command_parser.rb:12:in `parse'
	1: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse'
/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse': 809: unexpected token at '{ "commandType": "control", "command": {"command":"done"} }' (JSON::ParserError)
	9: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/swift_lane_manager.rb:223:in `block in start_socket_thread'
	8: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:38:in `start'
	7: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:158:in `listen'
	6: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `receive_and_process_commands'
	5: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:49:in `loop'
	4: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:62:in `block in receive_and_process_commands'
	3: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/socket_server.rb:81:in `parse_and_execute_command'
	2: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/fastlane-2.178.0/fastlane/lib/fastlane/server/command_parser.rb:12:in `parse'
	1: from /Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse'
/Users/redacted/Documents/Projects/redacted-ios/.vendor/ruby/2.6.0/gems/json-2.5.1/lib/json/common.rb:216:in `parse': [!] 809: unexpected token at '{ "commandType": "control", "command": {"command":"done"} }' (JSON::ParserError) </pre>
</details>

Where `test` is:

```swift
func testLane() {
    xcov()
}
```

### Environment

 _fastlane_ 2.178.0, the rest is irrelevant.

### PR DESCRIPTION
**Pull request by minuscorp**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Resolves #18433 

### Description

Make sure that when a `ConfigItem`'s `type` is set to a decimal point type take the default value and get its decimal point representation in case it is not yet valid.

### Testing Steps

1. Checkout: `https://github.com/minuscorp/fastlane/tree/swift-fix-decimal-precission-values`
2. Install it
3. Run `fastlane generate_swift_api`
4. Verify that floating point variables have a floating point default value:

image

### CODE REVIEW

File: fastlane/lib/fastlane/swift_fastlane_function.rb [hunk 1]
```diff
@@ -116,6 +116,7 @@ def override_default_value_if_not_correct_type(param_name: nil, param_type: nil,
     end
 
     def get_type(param: nil, default_value: nil, optional: nil, param_type_override: nil, is_string: true)
+      require 'bigdecimal'
```
Comment (rogerluan) [inline fastlane/lib/fastlane/swift_fastlane_function.rb:119]: Do we need this one over here?

**Review by rogerluan — COMMENTED** [review]
Thanks for picking up this issue to fix! 💪 

This implementation LGTM as long as you verified it works! Are there any unit tests that could be written to make sure this is working as expected? 😊 I'll run locally soon 🙏

**Comment by rogerluan** [comment]
I ran this branch locally and it works as expected (I also updated the PR description) 💪 it also looks like the first `require` can be safely removed 🤗

File: fastlane/lib/fastlane/swift_fastlane_function.rb (hunk 1 above)
Comment (joshdholtz) [reply fastlane/lib/fastlane/swift_fastlane_function.rb:119]: I _think_ we still want this here just for safety. `require bigdecimal` is required to use `BigDecimal`. I tested this out in `irb`. What is most likely happening is another library/dependency is also requiring `bigdecimal` which is why it works but we don’t _really_ want to rely on that 🤷‍♂️

**Review by joshdholtz — APPROVED** [review]
This is dope! Thanks for making this fix ❤️ 

And thanks for testing this, @rogerluan!

**Pull request merged**